tests/31sync/02sync.pl

```diff
@@ -1,3 +1,246 @@
+use Future::Utils qw( repeat );
+
+push our @EXPORT, qw(
+   matrix_do_and_wait_for_sync
+   matrix_send_room_text_message_synced
+   matrix_send_room_message_synced
+   matrix_create_room_synced
+   matrix_join_room_synced
+   matrix_leave_room_synced
+   matrix_invite_user_to_room_synced
+   matrix_put_room_state_synced
+   matrix_advance_room_receipt_synced
+   sync_timeline_contains
+);
+
+=head2 matrix_do_and_wait_for_sync
+
+   my ( $action_result ) = matrix_do_and_wait_for_sync( $user,
+      do => sub {
+         return some_action_that_returns_a_future();
+      },
+      check => sub {
+         my ( $sync_body, $action_result ) = @_
+
+         # return a true value if the sync contains the action.
+         # return a false value if the sync isn't ready yet.
+         return check_that_action_result_appears_in_sync_body(
+            $sync_body, $action_result
+         );
+      },
+   )->get;
+
+
+Does something and waits for the result to appear in an incremental sync.
+Doesn't affect the next_batch token used by matrix_sync_again.
+
+The do parameter is the a subroutine with the action to perform that returns
+a future.
+The check parameter is a subroutine that receives the body of an incremental
```

Comment:
"is the a"??

Also you might like to use `C<do>` etc.. to notate the names of params in fixedwidth font.